core/lib/types/src/contract_verification/contract_identifier.rs

```diff
@@ -90,19 +215,23 @@ impl ContractIdentifier {
         // Try to detect metadata.
         // CBOR takes precedence (since keccak doesn't have direct markers, so it's partially a
         // fallback).
-        let (detected_metadata, bytecode_without_metadata_keccak256) = if let Some((
-            full_length,
-            hash,
-        )) =
-            Self::detect_cbor_metadata(bytecode_marker, bytecode)
-        {
-            (Some(DetectedMetadata::Cbor { full_length }), hash)
-        } else if let Some(hash) = Self::detect_keccak_metadata(bytecode_marker, bytecode) {
-            (Some(DetectedMetadata::Keccak256), hash)
-        } else {
-            // Fallback
-            (None, bytecode_keccak256)
-        };
+        let (detected_metadata, bytecode_without_metadata_keccak256) =
+            if let Some((full_length, hash, cbor_metadata)) =
+                Self::detect_cbor_metadata(bytecode_marker, bytecode)
+            {
+                (
+                    Some(DetectedMetadata::Cbor {
```

Comment:
Nit: nesting makes it a bit too hard to read, I'd suggest assigning `DetectedMetadata` to a variable. Also, you can rename the variable to `metadata` for nicer constructing, e.g.

```rust
let detected_metadata = DetectedMetadata::Cbor {
  full_length,
  metadata
});
(detected_metadata, hash)
```